detect_secrets/util/importlib.py

```diff
@@ -0,0 +1,56 @@
+import pkgutil
+from importlib import import_module
+from types import ModuleType
+from typing import Any
+from typing import Callable
+from typing import Iterable
+from typing import Type
+
+
+def import_types_from_module(
+    root: ModuleType,
+    filter: Callable[[Any], bool],
+) -> Iterable[Type]:
+    output = []
+    modules = get_modules_from_package(root)
+
+    for module_path in modules:
+        module = import_module(module_path)
+        for name in dir(module):
+            if name.startswith('_'):
+                continue
+
+            attribute = getattr(module, name)
+            if filter(attribute):
+                continue
+
+            output.append(attribute)
+
+    return output
+
+
+def import_modules_from_package(
+    root: ModuleType,
+    filter: Callable[[str], bool],
+) -> Iterable[ModuleType]:
+    output = []
+    modules = get_modules_from_package(root)
+
+    # NOTE: It should be auto-sorted, but let's just do it for sanity sake.
```

Comment:
It's sorted so that we can perform upgrades in order. e.g. we don't want to run the code for `detect_secrets.core.upgrades.v1_0` before `detect_secrets.core.upgrades.v0_12`.

I added a comment to clarify this, since I wasn't immediately sure either when I read the code again 😅